components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/InfinispanComponent.java

```diff
@@ -69,7 +74,20 @@ protected Endpoint createEndpoint(String uri, String remaining, Map<String, Obje
         } else {
             conf = new InfinispanConfiguration();
         }
-        conf.setCacheContainer(cacheContainer);
+
+        String[] confParameters = new String[] {"hosts", "cacheContainerConfiguration", "configurationUri", "cacheContainer"};
+        //     init default embedded cache if config parameters aren't specified or cacheContainer is set using setMethod
+        if (!Arrays.stream(confParameters).anyMatch(parameters::containsKey) || setCacheFromComponent) {
```

Comment:
you probably need to check also the values set on the `InfinispanConfiguration`

Reply:
Missed this one.

Reply:
Thanks @lburgazzoli

Reply:
Is the `||` OR correct as the `setCacheFromComponent` is always true. So I think its is wrong, and it should be `&&` AND instead

Reply:
thanks. I didn't know about other possibilities of initialization. I added help method do determine whether configuration is provided or not.

Reply:
@davsclaus actually `setCacheFromComponent` is almost never true, at least in tests which I provided. Usually cacheContainer is initialized in `InfinispanManager` class.